File: src/Commands/Shoot.cpp
#include "Shoot.h"
#include <SmartDashboard/SmartDashboard.h>
#include "../OI.h"
Shoot::Shoot() {
	Requires(shooter.get());
	// Use Requires() here to declare subsystem dependencies
	// eg. Requires(Robot::chassis.get());
}

// Called just before this Command runs the first time
void Shoot::Initialize() {
	frc::SmartDashboard::PutBoolean("Shooting", true);
	shooter->IsShooting = true;
}

// Called repeatedly when this Command is scheduled to run
void Shoot::Execute() {
	shooter->SetFlywheel(CommandBase::oi->getFlywheelValue());
}

// Make this return true when this Command no longer needs to run execute()
bool Shoot::IsFinished() {
	return (robotTimer->Get()>15 && !OI::shootButton->Get());
}

// Called once after isFinished returns true
void Shoot::End() {
	shooter->SetFlywheel(0);
	frc::SmartDashboard::PutBoolean("Shooting", false);
	shooter->IsShooting = false;
}

// Called when another command which requires one or more of the same
// subsystems is scheduled to run
void Shoot::Interrupted() {
	shooter->SetFlywheel(0);
	frc::SmartDashboard::PutBoolean("Shooting", false);
	shooter->IsShooting = false;
}

File: src/org/usfirst/frc/team9514/robot/commands/Shoot.java
package org.usfirst.frc.team9514.robot.commands;

import edu.wpi.first.wpilibj.command.Command;
import org.usfirst.frc.team9514.robot.CommandBase;

/**
 *
 */
public class Shoot extends Command { 
    
    public Shoot() {
        // Use requires() here to declare subsystem dependencies
        // eg. requires(chassis);
        requires(CommandBase.shooter);
    }

    // Called just before this Command runs the first time
    protected void initialize() {
        CommandBase.shooter.enablePID();
    }

    // Called repeatedly when this Command is scheduled to run
    protected void execute() {
        double shooterOutput = CommandBase.shooter.getShooterPIDOutput();
        shooterOutput*=.5;//Scale
        if(CommandBase.oi.getShooter())
            CommandBase.shooter.setRaw(shooterOutput);
        else
            CommandBase.shooter.setRaw(0);
    }

    // Make this return true when this Command no longer needs to run execute()
    protected boolean isFinished() {
        return false;
    }

    // Called once after isFinished returns true
    protected void end() {
        CommandBase.shooter.disablePID();
        CommandBase.shooter.setRaw(0);
    }

    // Called when another command which requires one or more of the same
    // subsystems is scheduled to run
    protected void interrupted() {
        CommandBase.shooter.disablePID();
        CommandBase.shooter.setRaw(0);
    }
}
